packages/admin/admin/src/EditDialog.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button, createTheme } from "@mui/material";
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import * as React from "react";
+import { IntlProvider } from "react-intl";
+
+import { useEditDialog } from "./EditDialog";
+import { FinalForm } from "./FinalForm";
+import { Field } from "./form/Field";
+import { FinalFormInput } from "./form/FinalFormInput";
+import { MuiThemeProvider } from "./mui/ThemeProvider";
+import { RouterMemoryRouter } from "./router/MemoryRouter";
+
+describe("EditDialog", () => {
+    describe("onAfterSave", () => {
+        test("Should call onAfterSave when clicking save button", async () => {
+            const user = userEvent.setup();
+            const onAfterSave = jest.fn();
+
+            function Story() {
+                const [EditDialog, , editDialogApi] = useEditDialog();
+
+                return (
+                    <>
+                        <Button onClick={() => editDialogApi.openAddDialog()}>Open dialog</Button>
+                        <EditDialog onAfterSave={onAfterSave}>
+                            <FinalForm
+                                mode="edit"
+                                onSubmit={async () => {
+                                    // noop
+                                }}
+                            >
+                                <Field name="name" component={FinalFormInput} autoFocus />
+                            </FinalForm>
+                        </EditDialog>
+                    </>
+                );
+            }
+
+            const rendered = render(
+                <IntlProvider locale="en" messages={{}}>
+                    <MuiThemeProvider theme={createTheme()}>
+                        <RouterMemoryRouter>
+                            <Story />
+                        </RouterMemoryRouter>
+                    </MuiThemeProvider>
+                </IntlProvider>,
+            );
```

Comment:
I added a custom render (as suggested by the docs) here: cb3d7564b979275f81a2bd43e6ecf6d1b5c07ab5

What do you think? One minor concern: I needed to add the `paths` and `baseUrl` options to `tsconfig.json` (and not `tsconfig.test.json`) for IDE support. I doubt that this will break something (i.e. the build), but we should verify that.

Reply:
Fine by me. 

Since the pipeline works, the build should be okay, right? Or do you think this can affect the application build?

Reply:
> Since the pipeline works, the build should be okay, right?

Probably

> Or do you think this can affect the application build?

I don't think so. But we'll see once applications update to the latest version